aot/src/main/java/com/dylibso/chicory/aot/AotUtil.java

```diff
@@ -202,7 +202,7 @@ public static Class<?> jvmReturnType(FunctionType type) {
             case 1:
                 return jvmType(type.returns().get(0));
             default:
-                return Value[].class;
+                return long[].class;
```

Comment:
This will break as it's also used by host functions. Or do you plan to change `WasmFunctionHandle` as well?

Reply:
My plan was to keep the user API layer with the typed `Value[]` for the moment, do you think that we should re-evaluate?

I know this is, anyhow, a low level API, but asking ppl to deal with `long[]` seems a bit of a stretch.

Reply:
I’m suggesting to change it. Rather, if it is not changed, then this code will need to handle the two different styles of return type arrays.

Reply:
I'm going down the path to change everything to `long[]` in the lower level API.

The plan is to bridge the gap with the code-generation effort right after.

Reply:
@electrum I think I'm getting there, still a ~1300 are failing, if you have time to take a look, any hint or help is always appreciated :pray: